### PR TITLE
refactor(apport): split main into smaller parts

### PR DESCRIPTION
**Pull request by bdrung**
These refactoring are preparation work for https://github.com/canonical/apport/pull/290.

### CODE REVIEW

**Comment by bdrung** [comment]
IMO codecov can be ignored in this case.

**Review by schopin-pro — CHANGES_REQUESTED** [review]
A small nitpick and a naming issue.

File: data/apport [hunk 1]
```diff
@@ -966,6 +966,17 @@ def process_crash_from_kernel_with_proc_pid(
...
+    if not real_user.is_root() and is_closing_session(proc_pid, real_user):
+        logger.error("happens for shutting down session, ignoring")
+        return 0
+
+    # ignore systemd watchdog kills; most often they don't tell us the
+    # actual reason (kernel hang, etc.), LP #1433320
+    if is_systemd_watchdog_restart(options.signal_number, proc_pid):
+        logger.error("Ignoring systemd watchdog restart")
+        return 0
+
```
[the remark this answers is not in the document]
Comment (bdrung) [reply data/apport:985]: Valid point. Changed the commit message.

File: data/apport [hunk 2]
```diff
@@ -697,24 +697,7 @@ def parse_arguments(args: list[str]) -> argparse.Namespace:
...
-        options = parse_arguments(args)
-
-    if options.stop:
-        stop_apport()
-        return 0
-    if options.start:
-        start_apport()
-        return 0
-
+def _check_global_pid_and_forward(options: argparse.Namespace) -> bool:
```
[the remark this answers is not in the document]
Comment (bdrung) [reply data/apport:701]: That function is a mess. It does check and changes. I fail to find an easy way to split this function into parts that can be easily named. So I converted and extended the first comment into a docstring.

[the remark this answers is not in the document]
Comment (schopin-pro) [reply data/apport:701]: I'll take it!

**Pull request merged**